src/plugins/dataRefs/utils.ts

```diff
@@ -0,0 +1,120 @@
+import {DataSeriesExtended} from '../../types';
+
+export function integrate(timestamps: number[], values: DataSeriesExtended) {
+    if (timestamps.length < 2) {
+        return 0;
+    }
+    let t0 = timestamps[0];
+    let x0 = Number(values[0]);
+    let t1: number;
+    let x1: number;
+    let integral = 0;
+    for (let i = 1; i < timestamps.length; i++) {
+        x1 = Number(values[i]);
+        t1 = timestamps[i];
+
+        // we skip over all non-numeric values in a serie
+        // so that all holes and single points (surrounded by
+        // empty void) do not add anything to the integral
+        if (!Number.isNaN(x1) && !Number.isNaN(x0)) {
+            const dt = t1 - t0;
+            const dx = x1 - x0;
+            const area = ((x0 + dx / 2) * dt) / 1000; // convert milliseconds to seconds
```

Comment:
So the integral for

values = [1, 0, -1]
timestamps = [1, 2, 3]

will be 0


Is this expected value?

probably should use absolute values?